src/main/java/io/stargate/sgv2/jsonapi/util/JsonUtil.java

```diff
@@ -58,4 +63,47 @@ public static boolean equalsOrdered(JsonNode node1, JsonNode node2) {
     // For other nodes default equals() works fine:
     return Objects.equals(node1, node2);
   }
+
+  /**
+   * Helper method that may be used to see if given JSON Value looks like it is an EJSON-encoded
+   * value -- that is, JSON Object with 1 entry, name of which starts with "$" character -- without
+   * decoding value. No validation is done beyond feasibility: caller is likely to want to call
+   * other methods like {@link #extractEJsonDate} for further processing and validation/
+   *
+   * @param json JSON value to check
+   * @return {@code true} if value looks like it has to be EJSON value (valid or invalid); {@code
+   *     false} if not.
+   */
+  public static boolean looksLikeEJsonValue(JsonNode json) {
+    return json.isObject() && json.size() == 1 && json.fieldNames().next().startsWith("$");
```

Comment:
should we just have a getter here? `json.fieldNames().next()` -> `json.fieldNames().get(0)`, avoids creating iterator for a single value

Reply:
To use `get()` would need to know the name to look for, and here we want to see if that one key that exists starts with "$". I don't think there is an accessor that would give us that key without knowing it, or creating Iterator or equivalent.
Would be happy to avoid iterator if there's an alternate mechanism.